demo_nodes_cpp/src/topics/talker_loaned_message.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/register_node_macro.hpp"
+#include "rcutils/cmdline_parser.h"
+
+#include "std_msgs/msg/float64.hpp"
+#include "std_msgs/msg/string.hpp"
+
+#include "demo_nodes_cpp/visibility_control.h"
+
+using namespace std::chrono_literals;
+
+namespace demo_nodes_cpp
+{
+// Create a Talker class that subclasses the generic rclcpp::Node base class.
+// The main function below will instantiate the class as a ROS node.
+class LoanedMessageTalker : public rclcpp::Node
+{
+public:
+  DEMO_NODES_CPP_PUBLIC
+  explicit LoanedMessageTalker(const rclcpp::NodeOptions & options)
+  : Node("loaned_message_talker", options)
+  {
+    // Create a function for when messages are to be sent.
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    std::vector<std::string> args = options.arguments();
```

Comment:
Please remove logic related to argument parsing. It doesn't add any value to the demo since the topic can be changed via a remap. I am doing the same for the rest of the demos in https://github.com/ros2/demos/pull/401.

Reply:
done in https://github.com/ros2/demos/pull/394/commits/5f7dc32ceec8ff02805c82fa18ae719c627f57e3